### PR TITLE
[DROOLS-5828] executable-model test failure in test-compiler-integrat…

### PR DESCRIPTION
**Pull request by tkobayas**
…ion MatchTest

- lookup scoped var/declaration

**Ports** 
This PR is for 7.x

forward-port to main -> https://github.com/kiegroup/drools/pull/4059

**JIRA**: 
https://issues.redhat.com/browse/DROOLS-5828

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>

* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>

* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by tkobayas** [comment]
Hi @mariofusco , This PR contains 2 fixes for `MatchTest.testObjectsDeepOnNestedAccumulate`.

1. At executable-model generation, `$count` is not found at `AccumulateVisitor.nameExprParameter` because `RuleContext.definedVars` stores `$count_1_sCoPe` as both a key and a value. I think key has to be the original var name so stripped `_1_sCoPe`.
2. At runtime, `LambdaAccumulator$NotBindingAcc` is used for the accumulate function `$result : average($count)` then failed. That's strange (It needs a binding `$count`). So I added `KiePackagesBuilder.addInnerBindings` for this case so that the function can use `LambdaAccumulator$BindingAcc`. 

Please review. If you think it's a very edge case and you don't have time to review now, I can move this test to `EdgeCaseNonExecModelTest`.

Thanks!

File: drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/builder/generator/RuleContext.java [hunk 1]
```diff
@@ -366,7 +366,17 @@ public void addDeclarationReplacing(DeclarationSpec d) {
         }
         this.scopedDeclarations.put(d.getBindingId(), d);
         this.allDeclarations.put(d.getBindingId(), d);
-        definedVars.put(bindingId, bindingId);
+        String var = stripIfScoped(bindingId);
+        definedVars.put(var, bindingId);
+    }
+
+    private String stripIfScoped(String bindingId) {
+        if (bindingId.endsWith("sCoPe")) {
```
Comment (mariofusco) [inline drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/builder/generator/RuleContext.java]: That `"sCoPe"` string is a (unfortunately necessary) hack to give a scope to accumulate variables and prevent both name shading and its visibility outside the accumulate context. Your fix is ok, please just move that `"sCoPe"` into a `static final` constant String and replace any occurrence in our code accordingly.

Comment (tkobayas) [reply drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/builder/generator/RuleContext.java]: Sure, thanks!

**Review by mariofusco — APPROVED** [review]

**Review by hellowdan — APPROVED** [review]
